### PR TITLE
feat(frontend): Sidebar component mobile responsiveness bb-106

### PR DESCRIPTION
**Pull request by JD-Rosales**
### Summary
Make the sidebar component responsive.

### Added
- Menu icon

https://github.com/user-attachments/assets/241df8a6-bc5f-475c-b380-27b7a840d4d3

### CODE REVIEW

File: apps/frontend/src/libs/components/header/header.tsx [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
+			<div className={styles["menu-container"]}>
+				<input
+					checked={isSidebarOpen}
+					hidden
+					id="sidebar-toggler"
+					onChange={toggleSidebar}
+					type="checkbox"
+				/>
+				<label htmlFor="sidebar-toggler">
+					<img alt="Menu" className={styles["menu"]} src={menu} />
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/header/header.tsx:34]: Use existing Icon component

File: apps/frontend/src/libs/components/sidebar/sidebar.tsx [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
 	return (
-		<div className={styles["sidebar-container"]}>
+		<div
+			className={getValidClassNames(
+				styles["sidebar-container"],
+				isSidebarOpen && styles["open"],
+			)}
+		>
+			<label className={styles["close-btn"]} htmlFor="sidebar-toggler">
+				x
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/sidebar/sidebar.tsx:23]: Let's use icon here

File: apps/frontend/src/libs/components/header/header.tsx [hunk 3]
```diff
@@ -1,14 +1,40 @@
...
 	return (
 		<header className={styles["header"]}>
+			<div className={styles["menu-container"]}>
+				<input
+					checked={isSidebarOpen}
+					hidden
+					id="sidebar-toggler"
+					onChange={toggleSidebar}
+					type="checkbox"
+				/>
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/header/header.tsx:32]: Shouldn't be a checkbox input

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/app/app.tsx [hunk 4]
```diff
@@ -33,19 +39,32 @@ const App: React.FC = () => {
...
+						styles["app-container"],
+						isSidebarOpen && styles["open"],
+					)}
+				>
+					<Sidebar
+						isSidebarOpen={isSidebarOpen}
+						toggleSidebar={toggleSidebar}
+					/>
+					<div className={styles["main"]}>
+						<Header toggleSidebar={toggleSidebar} />
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/app/app.tsx:61]: A9

File: apps/frontend/src/libs/components/app/app.tsx [hunk 5]
```diff
@@ -33,19 +39,32 @@ const App: React.FC = () => {
 
 	const isLoading = dataStatus === DataStatus.PENDING;
 
+	const toggleSidebar = useCallback((): void => {
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/app/app.tsx:42]: A10

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/app/app.tsx (hunk 4 above)
Comment (JD-Rosales) [reply apps/frontend/src/libs/components/app/app.tsx:61]: I'm sorry, what does A9 means?

Comment (fshabanov) [reply apps/frontend/src/libs/components/app/app.tsx:61]: Rule number from [quality criteria](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/src/javascript.md)

**Review by WilhelmWesser — APPROVED** [review]

File: apps/frontend/src/libs/components/app/app.tsx [hunk 6]
```diff
@@ -33,19 +39,35 @@ const App: React.FC = () => {
 
 	const isLoading = dataStatus === DataStatus.PENDING;
 
+	const handleSidebarToggle = useCallback((): void => {
+		setIsSidebarOpen((previous) => !previous);
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/app/app.tsx:43]: `previous` is not a clear name

**Review by fshabanov — CHANGES_REQUESTED** [review]
Add a screenshot of mobile view as well

File: apps/frontend/src/libs/components/header/header.tsx [hunk 7]
```diff
@@ -1,14 +1,23 @@
...
+};
+
+const Header: React.FC<Properties> = ({ onSidebarToggle }: Properties) => {
 	const user: UserDto = useAppSelector(({ auth }) => auth.user as UserDto);
 
 	return (
 		<header className={styles["header"]}>
+			<button className={styles["menu-btn"]} onClick={onSidebarToggle}>
+				<Icon name="menu" />
+			</button>
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/header/header.tsx:19]: Use existing Button component. Add a possibility to add Icon there

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/button/button.tsx [hunk 8]
```diff
@@ -3,15 +3,17 @@ import { getValidClassNames } from "~/libs/helpers/helpers.js";
 import styles from "./styles.module.css";
 
 type Properties = {
+	icon?: React.ReactNode;
 	isFluid?: boolean;
 	isPrimary?: boolean;
-	label: string;
+	label?: string;
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/button/button.tsx:9]:
```suggestion
	label: string;
```
Label should be present. Add `hasVisuallyHiddenLabel` prop and if it is true, hide the label using `visually-hidden` class

File: apps/frontend/src/libs/components/header/header.tsx [hunk 9]
```diff
@@ -1,14 +1,27 @@
...
+};
+
+const Header: React.FC<Properties> = ({ onSidebarToggle }: Properties) => {
 	const user: UserDto = useAppSelector(({ auth }) => auth.user as UserDto);
 
 	return (
 		<header className={styles["header"]}>
+			<div className={styles["menu-btn"]}>
+				<Button
+					icon={<Icon name="menu" />}
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/header/header.tsx:19]:
```suggestion
					iconName="menu"
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
Merge the main branch into yours

**Pull request merged**